### ISSUE TITLE
why use new RxPermissions(this) instance of  RxPermissions.getInstance(this) ?

### ISSUE THREAD

**Issue by coolBreezes** [tbruyelle/RxPermissions#230]
Dear author,
I  see old api use RxPermissions.getInstance(this) to create a instance,
now use new to create a instance?
Why did you want to make this change?

Is it to prevent the destruction of the activity leading to fragment destruction?

**Comment by epool** [comment]
@coolBreezes This was a design change, with the old api `RxPermissions.getInstance(this)` we used to have a singleton static instance of `RxPermissions` with a linked `Application` context instance(the one passed by `this`.`getApplicationContext()`). This context was used to start a ghost activity for requesting permissions(a workaround to be less invasive for library user's apps). Then we opted to use a retained fragment to manage the permissions and get rid of the ghost activity and improve the UX without that ghost activity. This is pretty much the reason behind this api change, since we cannot longer manage a singleton instance, we let the library user's create/inject a singleton of `RxPermissions` with an `@ActivityScope` for example with Dagger. I hope I had answered your questions.
